### DISCUSSION TITLE
Nav won't animate on page change

### DISCUSSION THREAD

**Discussion by praetoriusd** [swup/swup#387] [Help] [answered]
I'm trying to get my nav to stay visible during page transitions but also change the "active" page to show that users are on a new page. When the link is clicked, nothing in the nav changes. 

For example, user is on the About page, clicks Contact. The nav will still show that the user is on the About page. 

I've read through all the documentation and cannot figure out how to animate this. 

my site is https://redeyedesign.ca 

Any guidance would be greatly appreciated.

**Comment by AugustMiller** [comment, marked as answer]
Hey! I don't see Swup implemented at the link, any more… so I don't know if we'll be able to troubleshoot.

Are you able to set up a couple of plain HTML files to demonstrate the problem, without needing to disrupt the live site?

Best guess: your navigation likely isn't included in one of the registered Swup `containers` ([docs](https://swup.js.org/options#containers)), and therefore doesn't update when navigating (assuming link highlighting is happening via class name changes on the server vs. dynamically via JS). You can either make your navigation element _another_ `container`, or 

The second step would be ensuring that the nav is not hidden (i.e. cross-faded) between pages. If you've applied some styles to whatever Swup container, expanding that container to include the nav may mean that *everything* downstream is affected.

Keep in mind that Swup does not intelligently update content of containers (or their child elements), so animating individual properties of the active link(s) may not be straightforward. For example, if you wanted to fade the underlines in-and-out, you'd need to cascade a the transition from the `.is-animating` class that Swup applies to `html`, remove all identifying styles, then reapply those styles after the new nav container is swapped in:

```css
nav a.current {
    color: red;
}

.is-animating nav a.current {
    color: inherit;
}
```
